File: app/src/main/java/com/example/daggerdemo/di/ActivityBuildersModule.java
package com.example.daggerdemo.di;

import com.example.daggerdemo.AuthActivity;

import dagger.Module;
import dagger.Provides;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuildersModule {

    @Provides
    static String returnStringObject() {
        return "returnSomeString()";
    }

    @ContributesAndroidInjector
    abstract AuthActivity contributesAuthActivity();

}

File: app/src/main/java/com/example/daggerdemo/di/ActivityBuildersModule.kt
package com.example.daggerdemo.di

import com.example.daggerdemo.AuthActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuildersModule {

    @ContributesAndroidInjector
    internal abstract fun contributesAuthActivity(): AuthActivity

}
